src/transformers/models/bart/modeling_bart.py

```diff
@@ -843,6 +843,30 @@ def __init__(self, config: BartConfig, embed_tokens: Optional[nn.Embedding] = No
 
         self.init_weights()
 
+    def get_input_embeddings(self):
+        return self.embed_tokens
+
+    def set_input_embeddings(self, value):
+        self.embed_tokens = value
+
+    def _prepare_decoder_attention_mask(self, attention_mask, input_shape, inputs_embeds, past_key_values_length):
```

Comment:
This both corrects a bug and refactors this rather complex function so that it can be copied into all Bart-ilke models